### ISSUE TITLE
Imperative VS. Declarative

### ISSUE THREAD

**Issue by wayjake** [adamjberg/coding-curriculum#2]
At first glance, _imperative may seem like the obvious choice_. "It does what you think it does," your mind tells you. It's what colleges teach, and that's because it's easier to teach that way. Your education as a programmer may suffer because of this one thing, so please listen up. Just because I said there are no wrong ways to program, doesn't mean I don't have some very strong opinions as to what code should look like. 

The key-phrase from above, is "**_I _think_ it does [this]_**." You will more often find yourself writing something and _think_ about why it's broken, but not be too sure. Declarative programming takes away this worry. It tends to be a more secure and testable format for composing code. The problem is that it's primarily a philosophy. 

Unlike so many things in software, this is one thing that may not be label-able. But how it's experienced is where you find yourself in the zone, and then you stumble upon some old code that isn't very clear about what it does, is probably imperative programming. 

From the dictionary, imperative means: of vital importance; crucial. I don't know what then lead programmers to name imperative because that doesn't seem related at all. 

[**Imparative programming_**](https://www.computerhope.com/jargon/i/imp-programming.htm#:~:text=Imperative%20programming%20is%20a%20paradigm,%22how%22%20to%20accomplish%20it.) is defined as: a paradigm of computer programming where the program describes steps that change the state of the computer.

Compared to **_declarative programming_**: Unlike declarative programming, which describes "what" a program should accomplish, imperative programming explicitly tells the computer "how" to accomplish it.

In other words, in the idea world, we aim to be the best at explaining _what_ I need to happen rather than explaining my exact solution. One holds a sign saying, "TURN RIGHT," and the other holds no sign knowing that the driver will know it's a turn and make it.

**Comment by wayjake** [comment]
An exercise here could be fun where they take a few function examples. Or you as the writer provides some "problems" that require code to solve.
